internal/server/openapi/api_model_registry_service_service.go

```diff
@@ -67,12 +70,17 @@ func (s *ModelRegistryServiceAPIService) CreateInferenceServiceServe(ctx context
 
 	result, err := s.coreApi.UpsertServeModel(entity, &inferenceserviceId)
 	if err != nil {
+		if errors.Is(err, api.ErrBadRequest) {
+			return Response(400, model.Error{Message: err.Error()}), nil
+		}
+		if errors.Is(err, api.ErrNotFound) {
+			return Response(404, model.Error{Message: err.Error()}), nil
+		}
```

Comment:
eventually, there can be a mapping function as discussed which for known-errors will return a `Response` if mapped or nil, so we can refactor similar code here.